modules/Emails/Email.php

```diff
@@ -4235,21 +4238,159 @@ protected function handleBodyInPlainTextFormat($mail)
         $this->description = $mail->Body;
     }
 
+    /**
+     * @param string $emailField
+     * @return string
+     */
+    public function getEmailAddressConfirmOptInTick($emailField)
+    {
+        global $sugar_config;
+        global $app_list_strings;
+        global $app_strings;
+        global $mod_strings;
+
+        $tickHtml = '';
+
+        if ($sugar_config['email_enable_confirm_opt_in']) {
+            $template = new Sugar_Smarty();
+            $template->assign('APP', $app_strings);
+            $template->assign('APP_LIST_STRINGS', $app_list_strings);
+            $template->assign('MOD', $mod_strings);
+            $template->assign('OPT_IN', $this->getEmailAddressOptInStatus($emailField));
+            $tickHtml = $template->fetch('include/SugarObjects/templates/basic/tpls/displayEmailAddressOptInField.tpl');
+        }
+
+        return $tickHtml;
+    }
+
     /**
      *
+     * @global array $sugar_config
+     * @global \LoggerManager $log
      * @param string $emailField
+     * @return \EmailAddress
+     * @throws RuntimeException
      * @throws InvalidArgumentException
      */
+    public function getEmailAddressConfirmOptIn($emailField)
+    {
+        global $sugar_config;
+
+        if (!$sugar_config['email_enable_confirm_opt_in']) {
+            global $log;
+            $log->warn('Confirm Opt In is not enabled.');
+
+            return false;
+        }
+
+        $emailAddressId = $this->getEmailAddressId($emailField);
+
+        return BeanFactory::getBean('EmailAddresses', $emailAddressId);
+    }
+
+    /**
+     * @param string $id
+     */
+    private function checkOptInFromEmailAddressId($id = '')
+    {
+        global $sugar_config;
+        global $log;
+
+        if ($id === '') {
+            $log->fatal('Empty Email Id');
+        }
+
+        $emailAddresses = BeanFactory::getBean('EmailAddresses');
```

Comment:
Performance: Please don't load the bean if you don't need it in this context - `$sugar_config['email_enable_auto_send_opt_in']` is false (see belove), even if the Email ID is empty (see above).